compiler/optimizer/VPHandlers.cpp

```diff
@@ -2704,21 +2704,6 @@ TR::Node *constrainIaload(OMR::ValuePropagation *vp, TR::Node *node)
             {
             TR_OpaqueClassBlock *clazz = vp->comp()->fej9()->getComponentClassFromArrayClass(constraint->getClass());
 
-            if (vp->comp()->getMethodHotness() >= scorching)
```

Comment:
How is this related to hot field marking? This is a different optimization I think... related to what nodes we should prefetch?

Reply:
This prefetch block is dependant on hot fields marked using the manor of marking hot fields. However upon further analysis, there is an underlying dependance here [https://github.com/eclipse/openj9/blob/6c0fa8e436ad4c87b0efb17b7731b86d12fb655f/runtime/compiler/p/codegen/J9CodeGenerator.cpp#L418]. @zl-wang, would this HotFieldPrefetch Power optimization be relevant and useful?

Reply:
How did you expect tenured alignment and padding has the same effect as hot object field prefetching? two different things to me, even if the end result to identify hot fields is the same eventually (which I am not sure only reading the comments).

In any case, my observation is hot field prefetching itself (instead of the iterator prefetching) was not close to a substantial performance booster though. the existing foundation might make it easier for further innovations in this area.

Reply:
Ok good to know . Thank you. I was not trying to say tenured alignment and padding has the same effect as hot object field prefetching, I was trying to say that those features are dependant on the old hot field infrastructure which I am trying to remove in this PR.  @zl-wang when you mention "not close to a substantial performance booster", what would be your thoughts towards its removal? If you feel strongly against removing it, should it be modified to rely on the new hot field marking so the rest of this PR can exist, or should it remain unchanged?

Reply:
ultimately, is the new mechanism able to identify same or better hot fields as the old infrastructure? (better in the sense that prefetching them makes a bigger difference of performance ... cache efficiency)
if it is not worse, i see no problem of hot-field prefetching piggy-back on the new identifications.

Reply:
Yes the new mechanism is able to better identify hot fields. I will then modify this PR so that hot field prefetching on Power relies on the new hot field infrastructure, and test this on Power against the the current hot field prefetching that relies on the old hot field instructure. Does this sound reasonable? @andrewcraik @zl-wang

Reply:
Sounds like a good solution to me - @zl-wang could you suggest some benchmarks to @jonoommen to be able to verify the change is working as designed?

Reply:
it sounds reasonable.  jbb2005 (not jbb2015) was the original targeted benchmark for these optimizations.  you might need -Xaggressive option to trigger.